source/dub/internal/temp_files.d

```diff
@@ -0,0 +1,45 @@
+/**
+	Provides methods to generate temporary file names and folders and
+	automatically clean them up on program exit.
+
+	Copyright: © 2012 Matthias Dondorff, © 2012-2023 Sönke Ludwig
+	License: Subject to the terms of the MIT license, as written in the included LICENSE.txt file.
+	Authors: Matthias Dondorff, Sönke Ludwig, Jan Jurzitza
+*/
+module dub.internal.temp_files;
+
+import std.file;
+
+import dub.internal.vibecompat.core.file;
+
+NativePath getTempDir()
+{
+	return NativePath(std.file.tempDir());
+}
+
+NativePath getTempFile(string prefix, string extension = null)
+{
+	import std.uuid : randomUUID;
+	import std.array: replace;
+
+	string fileName = prefix ~ "-" ~ randomUUID.toString() ~ extension;
+
+	if (extension !is null && extension == ".d")
+		fileName = fileName.replace("-", "_");
```

Comment:
I think this has the effect of modifying the `prefix` in case it continues `-`, which could lead to weird behavior (especially if the callers of this function are not prepared to handle this case and perform other commands relative to the prefix, without substituting `-` for `_`.
How about either:
- only modifying the baseName component, and not the dirName?
- always using `prefix ~ '_' ~ ...` no matter the file extension

Reply:
This file is just moved from the utils module into its own to fix cyclic dependency with module dtors. No behavior is changed and this function has already been widely used and tested. If you have some problem with its behavior, please change it in a separate PR